client/js/controllers/post_main_controller.js

```diff
@@ -41,16 +41,29 @@ class PostMainController extends BasePostController {
                     router.replace(url, ctx.state, false);
                 }
 
+                const prevPostId = aroundResponse.prev
+                        ? aroundResponse.prev.id
+                        : null;
+                const nextPostId = aroundResponse.next
+                        ? aroundResponse.next.id
+                        : null;
+
+                // preload nearby posts
+                if (prevPostId) {
+                    Post.get(prevPostId, { noProgress: true }).then(misc.preloadPostImages);
+                    PostList.getAround(prevPostId, parameters ? parameters.query : null, { noProgress: true });
+                }
+                if (nextPostId) {
+                    Post.get(nextPostId, { noProgress: true }).then(misc.preloadPostImages);
+                    PostList.getAround(nextPostId, parameters ? parameters.query : null, { noProgress: true });
```

Comment:
I can't quite understand why we're calling the `getAround` endpoint in this scenario – can you please help me understand? Thanks

Reply:
Rendering is blocked until `getAround` returns, so we need to have it in cache for instant navigation.
https://github.com/rr-/szurubooru/blob/e3794352f332e0ac09eba30bd5c0ce06337ac76a/client/js/controllers/post_main_controller.js#L20-L28